minari/data_collector/data_collector.py

```diff
@@ -209,13 +209,19 @@ def step(
         # This may happen if the step_data_callback truncates or terminates the episode under certain conditions.
         if self._new_episode and not self._reset_called:
             self._buffer[-1]["observations"] = [self._previous_eps_final_obs]
+            if self._record_infos:
+                self._buffer[-1]["infos"] = self._add_to_episode_buffer(
+                    {}, self._previous_eps_final_info
+                )
             self._new_episode = False
 
         # add step data to last episode buffer
         self._buffer[-1] = self._add_to_episode_buffer(self._buffer[-1], step_data)
 
         if step_data["terminations"] or step_data["truncations"]:
             self._previous_eps_final_obs = step_data["observations"]
+            if self._record_infos:
```

Comment:
Can you add a small comment on why we are saving the last obs and info?

Reply:
Done. Also updated the comment above to reflect the fact that infos are also being saved.